### DISCUSSION TITLE
Can we have speech_segments or even silence_segments in seconds in TranscriptionInfo?

### DISCUSSION THREAD

**Discussion by guilhermehge** [SYSTRAN/faster-whisper#481] [Ideas]
Since we already have it as a DEBUG log, I believe it would be a great addition to the TranscriptionInfo so we can access it easily. I believe that the best way to have that is by dividing the segments by the sampling rate, so that we have it in the seconds.

For silence_segments, we would need to build a function that iterate over those segments and returns what is not speech.

What do you guys think? I believe that the speech segments is of easy implementation.

**Comment by toanhuynhnguyen** [comment]
When calling the function model.transcribe, you can use these arguments:
```batch
vad_filter=True,
vad_parameters={
    "threshold": 0.2,
    "min_speech_duration_ms": 10,
    "min_silence_duration_ms": 1000,
    "speech_pad_ms": 400,
},
```

You'll be able to tune these arguments on your demand.

**Comment by guilhermehge** [comment]
This idea is not about tuning the parameters, it is about having access to what the model did afterwards. The VAD model can output the silence segments it detected with keys "start" and "end".

For example:
If we have silence segments being an output of the transcriptions, we'd have access to a list of dicts:
Silence segment 1:
{[
    "start": 0.455,
    "end": 2.113
 },{
    "start": 3.443,
    "end": 4.332
 }
]

With this I could calculate the amount of silence and even get the opposite of it, that is, speech segments.

**Reply by MahmoudAshraf97** [reply]
the `TranscriptionInfo` class already has `duration_after_vad`

**Reply by guilhermehge** [reply]
Yes, but that's just a full number that we can subtract from the full duration, the information that we can get from that is a simple "30s of silence" for example. What I meant is having each segment that was speech, with timestamps and all, so I can know that at time from x to y, it was silent. This information is already available, but not accessible, since it is not added to the TranscriptionInfo class.

It is what I proposed in this pull request that was closed.

https://github.com/SYSTRAN/faster-whisper/pull/487
